### PR TITLE
cleaning up existing dataset test cases

### PR DESCRIPTION
**Pull request by hhsecond**
## Description
- Cleaning up existing dataset test cases
- Adding more Variablesized Dataset tests

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **[CONTRIBUTING](../CONTRIBUTING.rst)** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by hhsecond** [comment]
> 1. adding huge amounts of samples (over 100k in a dataset)
> 2. adding huge amounts of datasets (>50)
> 3. iterating over datasets and dataset samples with the `keys()`, `values()`, and `items()` methods.
> 4. explicitly testing the dictionary style set/get/delete methods (to ensure the corresponding `"dunder"` methods are working) separately from the named `add()`, `get()`, and `remove()` methods.

I would add 1, 2 & all other stress testing stuff to another PR and keep those as separate test cases so that if people want to skip those while running tests. Will add others to this PR itself

File: tests/test_dataset.py [hunk 1]
```diff
@@ -138,19 +143,25 @@ def remove_all_data(self, written_repo, array5by7):
         co.close()
         co = written_repo.checkout(write=True)
         co.datasets['_dset'].remove('1')
-        # TODO: deleting all elementes supposed to delete dset - Not happening
-        # TODO: adding another element and deleting it again raises
-        # AttributeError: 'NoneType' object has no attribute 'decode'
+        # TODO: removing all datasets removes indexing. test that later
```
[the remark this answers is not in the document]
Comment (hhsecond) [reply tests/test_dataset.py]: Actually, I did not read it enough and kept it for later. Found this [here](https://github.com/tensorwerk/hangar-py/blob/044937c5a00fa48bd2889cce878648404645591d/src/hangar/dataset.py#L497)

**Comment by hhsecond** [comment]
@rlizzo This can be merged

**Comment by rlizzo** [comment]
Looks good. Merging now.

**Pull request merged**